cmd/kubelet/app/server.go

```diff
@@ -1296,3 +1303,24 @@ func parseResourceList(m map[string]string) (v1.ResourceList, error) {
 	}
 	return rl, nil
 }
+
+func checkCriStatus(runtimeService internalapi.RuntimeService, r *http.Request) error {
+	status, err := runtimeService.Status()
+	if err != nil {
+		return err
+	}
+	if _, found := r.URL.Query()["ignore_conditions"]; found {
+		return nil
+	}
+	var messages []string
+	for _, c := range status.GetConditions() {
+		if c.GetType() != "RuntimeReady" || c.GetType() != "NetworkReady" {
+			continue
+		}
+		messages = append(messages, fmt.Sprintf("%s=%t reason:%s message:%s", c.GetType(), c.GetStatus(), c.GetReason(), c.GetMessage()))
+	}
+	if len(messages) > 0 {
+		return fmt.Errorf("Runtime Conditions: %s", strings.Join(messages, ", "))
```

Comment:
I might be misunderstanding but shouldn't we check that `RuntimeReady` condition and say `NetworkReady` condition are both true, and if so return nil (no error). It seems like we always will return error here with the conditions and I assumed error should only be returned if it failed the health check?

Reply:
yeah, you're right. Changed.